src/bridges/usdcBridge.ts

```diff
@@ -0,0 +1,283 @@
+import {
+  BigNumberish,
+  BytesLike,
+  ethers,
+  AbiCoder,
+  ContractTransactionResponse,
+  Overrides,
+} from 'ethers';
+import {
+  Address,
+  PaymasterParams,
+  PriorityOpResponse,
+  TransactionResponse,
+} from '../types';
+import {IL1Nullifier__factory} from '../typechain';
+import {
+  REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT,
+  checkBaseCost,
+  scaleGasLimit,
+} from '../utils';
+import {Wallet} from '../wallet';
+import {insertGasPrice} from '../adapters';
+
+interface IDepositTransaction {
+  token: Address;
+  amount: BigNumberish;
+  to?: Address;
+  operatorTip?: BigNumberish;
+  bridgeAddress: Address;
+  approveERC20?: boolean;
+  l2GasLimit?: BigNumberish;
+  gasPerPubdataByte?: BigNumberish;
+  refundRecipient?: Address;
+  overrides?: Overrides;
+  approveOverrides?: Overrides;
+}
+
+interface IWithdrawTransaction {
+  token: Address;
+  amount: BigNumberish;
+  to?: Address;
+  bridgeAddress: Address;
+  paymasterParams?: PaymasterParams;
+  overrides?: Overrides;
+  approveERC20?: boolean;
+  approveOverrides?: Overrides;
+}
+
+export class USDCBridge {
```

Comment:
1. Can you add a doc-comment explaining that this bridge is for native USDC bridge (with link) and explain the difference with `USDC.e`?
2. Given that we probably will support more bridges in the future, maybe it makes sense to implement a class/interface for abstract bridge and make this class implement it?

Reply:
1. I've added a doc with a link to the USDC repo.
2. Good idea, added it.